pallets/capacity/src/lib.rs

```diff
@@ -194,6 +194,8 @@ pub mod pallet {
 		type EraLength: Get<u32>;
 
 		/// The maximum number of eras over which one can claim rewards
+		/// Note that you can claim rewards even if you no longer are boosting, because you
+		/// may claim rewards for past eras up to the history limit.
```

Comment:
This is the same as example 4.  My understanding is that no, she can NOT claim that reward.  She has 10 eras to claim it and then the ability to claim rewards expires.  If she tries, then that is when the (future) code in the claim extrinsic would compare it to the history limit, and delete the record since not only is there nothing staked but what she did stake is far beyond the history limit.

Reply:
Is that bad from the cleanup perspective? Alice has no reason to make the call that would cleanup the out of date history.

Or perhaps a more general question, why not allow it to be claimed? Or is the issue we don't want to retain the information about prior total staking amounts in `StakingRewardPool`, however I don't think that is currently getting any data removed.

Reply:
The problem is the overall RewardPool history retention.  We can't keep that history around forever.  Without that, there's no way of calculating old rewards.  If Alice has no reason to make the call that would clean up the out of date history, it's possible there could be some periodic check that will do it, perhaps run manually, or even on a runtime upgrade.

However I'm curious as to how much of a problem you anticipate that would be since it's literally in people's interest to claim their rewards before they expire.

Reply:
True, I'm mostly just thinking through consequences, but I see as you say that `start_new_reward_era_if_needed` has the removal, so I had just missed that piece. All good, thanks!